File: 242-ValidAnagram/242-ValidAnagram.c
// Last updated: 3/30/2026, 2:13:32 PM
bool isAnagram(char* s, char* t) {
    int freq[256] = {0};
    for(int i = 0 ; i < strlen(s) ; i++){
        freq[s[i]]++;
    }
    for(int i = 0 ; i < strlen(t) ; i++){
        freq[t[i]]--;
    }
    for(int i = 0 ; i < 127 ; i++){
        if(freq[i]!=0) return false;
    }
    return true;
}

File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 3/30/2026, 2:13:37 PM
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer> map = new HashMap<>();
        for(char ch: s.toCharArray()) map.put(ch,map.getOrDefault(ch,0)+1);
        for(char ch: t.toCharArray()) map.put(ch,map.getOrDefault(ch,0)-1);
        for(int value : map.values()){
            if(value!=0) return false;
        }
        return true;
    }
}